schemas/hygrothermalData.json

```diff
@@ -1,37 +1,183 @@
-            },
-            "dataFormat": {
-              "$ref": "common.json#/$defs/webAddress",
-              "description": "The web address provides a definition of the format of the 'xmlData'."
-            },
-            "xmlData": {
-              "$ref": "string.json#/$defs/extensibleMarkupLanguageDocument",
-              "description": "Hygrothermal data in Extensible Markup Language (XML) format in the exact data format specified by the property 'formatting'."
+        "thermalConductivityDry": {
+          "$ref": "number.json#/$defs/wattPerMeterKelvin"
+        },
+        "bulkDensity": {
+          "$ref": "number.json#/$defs/kilogramPerCubicMeter",
+          "description": "short description"
+        },
+        "porosity": {
+          "$ref": "number.json#/$defs/percentDecimalBetweenZeroAndOne",
+          "description": "short description"
+        },
+        "specificHeatCapacity": {
+          "$ref": "number.json#/$defs/joulePerKgK",
+          "description": "short description"
+        },
+        "waterVaporDiffusionResistanceFactor": {
+          "$ref": "number.json#/$defs/nonNegativeNumber",
+          "description": "short description"
+        },
+        "referenceWaterContent": {
+          "$ref": "number.json#/$defs/kilogramPerCubicMeter",
+          "description": "short description"
+        },
+        "freeWaterSaturation": {
+          "$ref": "number.json#/$defs/kilogramPerCubicMeter",
+          "description": "short description"
+        },
+        "waterAbsorptionCoefficient": {
+          "$ref": "number.json#/$defs/kilogramPerSquareMeterAndSquareRootSeconds",
+          "description": "short description"
+        },
+        "moistureDependentConductivitySupplement": {
+          "$ref": "number.json#/$defs/percentPerMassPercent",
+          "description": "short description"
+        },
+        "temperatureDependentConductivitySupplement": {
+          "$ref": "number.json#/$defs/wattPerMeterSquareKelvin",
+          "description": "short description"
+        },
+        "typicalBuildInMoisture": {
+          "$ref": "number.json#/$defs/kilogramPerCubicMeter",
+          "description": "short description"
+        },
+        "moistureStorageFunction": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "relativeHumidityAbsolut": {
+                "$ref": "number.json#/$defs/percentDecimalBetweenZeroAndOne",
+                "description": "short description"
+              },
+              "waterContent": {
+                "$ref": "number.json#/$defs/kilogramPerCubicMeter",
+                "description": "short description"
+              }
             }
-          },
-          "additionalProperties": false,
-          "minProperties": 1,
-          "required": ["xmlData"]
+          }
+        },
+        "moistureStorageFunctionGenerated": {
```

Comment:
This allows the unclear situation that the "moistureStorageFunction" is provided in detail while "moistureStorageFunctionGenerated" is false. You can change it to a "oneOf" where either the moistureStorageFunction is provided or the value is "not_provided". Strings can more precise than booleans. Please change the other arrays accordingly.

Reply:
- "Functions" can be defined by the array (but not required) 
- or they can be genenerated (not required) - but if generated with addtional parameters
![Screenshot 2024-07-08 140742](https://github.com/building-envelope-data/api/assets/169238265/1edc9160-e64e-461d-b516-09322fd2b91a)

Reply:
not sure the best way how to define this

Reply:
@c3rromatthias Have you decided yet if you want to use "oneOf"? It would also be possible to define the method which was used to create the array by the uuid of the method in https://www.buildingenvelopedata.org/methods . You can easily register all relevant methods there.

Reply:
We think that "oneOf" cannot capture the complex relations between the keys.